dotaservice/lua/hero_selection.lua

```diff
@@ -16,25 +61,67 @@ function GetBotNames ()
     }
 end
 
-local bot_heroes = {"npc_dota_hero_nevermore"}
+local function mode1v1()

```

Comment:
Why are there per-mode picks?

Reply:
I did it this way as if we ever want to extend to non AllPick game modes we will have the flexibility.

Captain's Mode - the traditional tournament mode is - has Ban, Pick, Ban, Pick, Ban, Pick sequence which is not present in All Pick for example. Hence why the logic would have to be different. Additionally, only in Lobby's can you pick all the heroes at once and not worry about pick order.

Reply:
If we generalize abilities enough we could support Ability Draft game modes too which would be super amazing.

Reply:
Well yeah but its often easier to work in small chunks of work at a time to unblock. If you get this working it'd amazing though. It will involve a bunch of back and forth between dotaservice and the agent.

Reply:
my worry is that I don't think the protobuf messages during hero selection work properly though. In all my testing a year ago they did not notify you of hero selections performed thus far... all the protobuf parameters were empty..... :(

I need to see if that's been fixed. If not, might be a good question to ask OpenAI since they had it working.

Reply:
I think OpenAI has their own addon